build.gradle

```diff
@@ -137,21 +136,42 @@ spotbugsTest {
     }
 }
 
-// task: pull default image
-task pullDockerImage(type: Exec) {
-    workingDir '.'
-    commandLine 'docker', 'pull', dockerGcrHostname + '/' + dockerGcrProject + '/' + dockerImageName + ':' + dockerImageTag
+// plugin: distribution
+distributions {
+    main {
+        distributionBaseName = 'terra-cli'
+        // configure what to include in the release archive
+        contents {
+            from 'tools/install.sh'
+            from 'README.md'
+        }
+    }
 }
 
-// task: build image
-task buildDockerImage(type: Exec) {
-    workingDir '.'
-    commandLine 'sh', 'tools/build-docker.sh', dockerLocalImageName
+// plugin: application
+application {
+    mainClass = 'bio.terra.cli.command.Main'
+    applicationName = 'terra'
+    executableDir = 'bin'
 }
-
-// task: publish image
-// expect caller to override the local tag property on the command line (-PdockerLocalImageTag=b5fdce0)
-task publishDockerImage(type: Exec) {
-    workingDir '.'
-    commandLine 'tools/publish-docker.sh', dockerLocalImageName, dockerLocalImageTag, dockerGcrHostname + '/' + dockerGcrProject + '/' + dockerImageName, dockerLocalImageTag
+startScripts {
+    // set the $APP_HOME path to $HOME/.terra
+    doLast {
+        // add the -PforRelease flag to update the APP_HOME in preparation for install
+        // default is to not update APP_HOME, so that ./gradlew install still works for development
+        if(project.hasProperty("forRelease")) {
+            println("Updating APP_HOME")
+            unixScript.text = unixScript.text.replace('APP_HOME="`pwd -P`"', 'APP_HOME="$HOME/.terra"')
```

Comment:
Yes, I thought the same thing, but couldn't figure out how to do that without writing a separate script. This script is generated by Gradle, so I can't modify it directly. We could write our own script from scratch, but the Gradle version is nice because it builds the Java classpath from the build.gradle dependencies -- we'd have to replicate that. We could also write a second script that calls the Gradle script, and then install both, but that seems messier to me. I'd be up for investigating other options, but I think it would make sense to do that in a follow-on ticket instead.

Reply:
Definitely. Thanks for the exmplanation. Also, is this supported on Windows? I didn't see any batch files (though I hear newer versions of Windows have bash?).

Reply:
The download/install scripts that I wrote are Bash-only, so those would have to be ported. Gradle generates both a Windows and *nix script, so theoretically the actual code should "just work" on both platforms. I've not tested this though, and have high suspicions that we'll need to do at least something to make it work.

RE: newer version of Windows. I'd not heard that, very interesting! In the past, I've used Cygwin in order to run Bash scripts on Windows.